metricbeat/beater/event.go

```diff
@@ -48,21 +48,32 @@ func (b eventBuilder) build() (common.MapStr, error) {
 		}
 	}
 
-	event = common.MapStr{
-		"@timestamp": timestamp,
-		"type":       typeName,
+	// Checks if additional meta information is provided by the MetricSet under the key _module
+	// This is based on the convention that each MetricSet can provide module data under the key _module
+	moduleData, moudleDataExists := event["_module"]
+	if moudleDataExists {
+		delete(event, "_module")
+	}
 
+	event = common.MapStr{
+		"@timestamp":            timestamp,
+		"type":                  typeName,
 		common.EventMetadataKey: b.metadata,
-		b.moduleName: common.MapStr{
-			b.metricSetName: event,
+		b.ModuleName: common.MapStr{
+			b.MetricSetName: event,
 		},
 		"metricset": common.MapStr{
-			"module": b.moduleName,
-			"name":   b.metricSetName,
-			"rtt":    b.fetchDuration.Nanoseconds() / int64(time.Microsecond),
+			"module": b.ModuleName,
+			"name":   b.MetricSetName,
+			"rtt":    b.FetchDuration.Nanoseconds() / int64(time.Microsecond),
 		},
 	}
 
+	// In case meta data exists, it is added on the module level
+	if moudleDataExists {
+		event[b.ModuleName].(common.MapStr).Update(moduleData.(common.MapStr))
```

Comment:
I was referring to testing that `moduleData` is a common.MapStr. I didn't see where that was validated.